packages/base/source/adt/union/union.js

```diff
@@ -82,8 +108,15 @@ instead to check if a value belongs to the ADT variant.`);
        *   where 'b = 'a[`@@folktale:adt:tag]
        */
       matchWith(pattern) {
-        return pattern[name](this);
-      } 
+        assertObject('${name}#matchWith', pattern);
```

Comment:
Ah, this should use template strings, as JS doesn't do interpolation on regular ones.

    `${name}#matchWith`